table/table.go

```diff
@@ -78,7 +78,7 @@ type Table interface {
 	Truncate(ctx context.Context) (err error)
 
 	// AddRecord inserts a row into the table.
-	AddRecord(ctx context.Context, r []interface{}) (recordID int64, err error)
+	AddRecord(ctx context.Context, r []interface{}, h int64) (recordID int64, err error)
```

Comment:
@qiuyesuifeng 
I think 0 is a better choice here, as 0 is both the default value in Go and invalid recordID. In addition, in some cases the recordID will be set to `LastInsertID`, which is an `uint64`, we would have to take special care of the casting if the value could be -1.